src/gameplay/PastMoves.js

```diff
@@ -1,68 +1,96 @@
-import {Card} from "react-bootstrap";
-import {RoomContext} from "../contexts/RoomContext";
-import {firestore, root} from "../config/firebase";
+import React, { useContext } from 'react';
+import { Card, ListGroup } from "react-bootstrap";
+import { RoomContext } from "../contexts/RoomContext";
+import { firestore, root } from "../config/firebase";
 
 function PastMoves() {
     const [moves, setMoves] = React.useState([]);
-    
     const { roomName } = useContext(RoomContext);
-    
+
     React.useEffect(() => {
-        const unsubscribe = firestore.collection(root).doc(roomName).collection("turns").onSnapshot((docs) => {
-            let moves = Array(docs.length);
-            docs.forEach(function (doc) {
-                const turn = doc.data();
-                const turn_num = parseInt(doc.id);
-                const move = turn.move;
-                let move_msg = "";
-                switch (turn.move.type) {
-                    case "general_income":
-                        move_msg = `${turn.playerName} took general income`;
-                        break;
-                    case "foreign_aid":
-                        move_msg = `${turn.playerName} took foreign aid`;
-                        break;
-                    case "duke":
-                        move_msg = `${turn.playerName} took 3 coins as Duke`;
-                        break;
-                    case "exchange_cards":
-                        move_msg = `${turn.playerName} exchanged cards as Ambassador`;
-                        break;
-                    case "steal":
-                        move_msg = `${turn.playerName} stole 2 coins from ${move.to}`;
-                        break;
-                    case "assassinate":
-                        move_msg = `${turn.playerName} assassinated ${move.to}`;
-                        break;
-                    case "coup":
-                        move_msg = `${turn.playerName} launched a coup on ${move.to}`;
-                        break;
-                    default:
-                        alert("Invalid move type");
-                        break;
-                }
-                moves[turn_num] = move_msg;
+		const subscribe = firestore.collection(root).doc(roomName).onSnapshot((roomDoc) => {
+            let currentTurn = roomDoc.data().turn;
+            roomDoc.ref.collection("turns").get().then((docs) => {
+                let moves = Array(docs.length);
+                docs.forEach(function (doc) {
+                    const turn = doc.data();
+                    const turn_num = parseInt(doc.id);
+                    const move = turn.move;
+                    let move_msg = [];
+                    console.log(currentTurn);
+                    if(currentTurn !== turn_num){
+                        switch (turn.move.type) {
+                            case "general_income":
+                                move_msg.push(`${turn.playerName} took general income`);
```

Comment:
Is it possible to say “attempting” and then “took” once the turn has been confirmed or blocked

Reply:
Oh Good idea, then we dont need to rely as heavily on waiting messages to be accurate, all players can see the current turn as it happens. I will work on it later tonight.